File: python/python_fundamentals/functions/functions_exercise/password_validator.py
def characters_long(passwd):
    if 6 <= len(passwd) <= 10:
        return True

    print("Password must be between 6 and 10 characters")
    return False

def special_char(passwd):
    if passwd.isalnum():
        return True

    print("Password must consist only of letters and digits")
    return False


def number_check(passwd):
    count_number = 0

    for number in passwd:
        if number.isdigit():
            count_number += 1
        if count_number >= 2:
            return True

    print("Password must have at least 2 digits")
    return False


some_password = str(input())

validator = characters_long(some_password), number_check(some_password), special_char(some_password)

if all(validator):
    print("Password is valid")

File: Front-End JS/JS Front-End/FunctionsAndStatementsExercises/PasswordValidator.js
function solve(password) {
    if (lengthValidator(password) && isLetterDigit(password) && numCountMinimum(password)) {
        console.log('Password is valid');
    } else {
        if (!lengthValidator(password)) {
            console.log('Password must be between 6 and 10 characters')
        }
        if (!isLetterDigit(password)) {
            console.log('Password must consist only of letters and digits');
        }
        if (!numCountMinimum(password)) {
            console.log(`Password must have at least 2 digits`);
        }
    }

    function isDigit(num) {
        return num >= 48 && num <= 57;
    }

    function numCountMinimum(password) {
        let count = 0;
        // let isDigit = (x) => x >= 48 && x <= 57;

        for (let char of password) {
            let charValue = char.charCodeAt(0);
            if (isDigit(charValue)) {
                count++;
            }
        }

        return count >= 2;
    }

    function lengthValidator(password) {
        return password.length >= 6 && password.length <= 10;
    }

    function isLetterDigit(password) {
        let isLowerLetter = (x) => x >= 87 && x <= 122;
        let isUpperLetter = (x) => x >= 65 && x <= 90;

        for (let char of password) {
            let charValue = char.charCodeAt(0);
            if (!isDigit(charValue) && !isLowerLetter(charValue) && !isUpperLetter(charValue)) {
                return false;
            }
        }

        return true;
    }
}